server/src/main/java/io/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -28,6 +28,14 @@
  */
 public interface IndexerMetadataStorageCoordinator
 {
+  /**
+   * Get all segments which may include any data in the interval. Any job using this method to do a merge should also hold
+   * a lock on the interval to prevent race conditions in other jobs.
+   * @param dataSource The datasource to query
+   * @param interval The interval over which to query for segments which should be included in a merge. Start is inclusive, end is exclusive
```

Comment:
it seems a bit strange to talk about merging here given that the method is used by other tasks as well.
As long as it's clear that segments returned may extend beyond the given interval, I don't think we need to talk about locking or merging here, but we should certainly put a comment in the code that uses those methods.